sentry-log4j/src/main/java/io/sentry/log4j/SentryAppender.java

```diff
@@ -47,7 +47,7 @@
      * <p>
      * Might be null in which case the DSN should be detected automatically.
      */
-    protected String dsn;
+    protected Dsn dsn;
```

Comment:
The diff here is in all 4 appenders. It makes more sense to do lookup at client creation time (in the factory). This also explains some changes in the tests (since appenders no longer do their own Dsn lookups).